app/services/voucher_adjustments_service.rb

```diff
@@ -85,6 +83,14 @@ def update_tax_adjustment_for(adjustment, tax_amount)
     # Update the amount if tax adjustment already exist, create if not
     tax_adjustment = @order.adjustments.find_or_initialize_by(adjustment_attributes)
     tax_adjustment.amount = tax_amount
+
+    # Add metada so we know which voucher adjustment is Tax related
+    tax_adjustment.metadata ||= AdjustmentMetadata.new(
+      enterprise: adjustment.originator.enterprise,
```

Comment:
This only gets set once, when the tax adjustment is initialised, so I wondered if this value could get out of date. But I don't think so: I expect the order/voucher enterprise can't change.